bofire/strategies/predictives/acqf_optimization.py

```diff
@@ -537,151 +516,112 @@ def _setup_ask(
-                            CategoricalEncodingEnum.DESCRIPTOR
-                        )
-                        for j, idx in enumerate(features2idx[feat.key]):
-                            if lower[j] == upper[j]:
-                                fixed_features[idx] = lower[j]
-        return fixed_features
+        elif optimizer == "optimize_acqf_list":
+            n_combos = domain.inputs.get_number_of_categorical_combinations()
+            return _OptimizeAcqfListInput(
+                acq_function_list=acqfs,
+                bounds=bounds,
+                num_restarts=self.n_restarts,
+                raw_samples=self.n_raw_samples,
+                options=self._get_optimizer_options(domain),  # type: ignore
+                inequality_constraints=inequality_constraints,
+                equality_constraints=equality_constraints,
+                nonlinear_inequality_constraints=nonlinear_constraints,
+                ic_generator=ic_generator,
+                ic_gen_kwargs=ic_gen_kwargs,
+                fixed_features_list=self.get_categorical_combinations(domain)
+                if n_combos > 1
+                else None,
+                fixed_features=self.get_fixed_features(domain=domain)
+                if n_combos == 1
+                else None,
+            )
+        elif optimizer == "optimize_acqf_mixed_alternating":
+            fixed_keys = domain.inputs.get_fixed().get_keys()
+            return _OptimizeAcqfMixedAlternatingInput(
+                acq_function=acqfs[0],
+                bounds=bounds,
+                q=candidate_count,
+                num_restarts=self.n_restarts,
+                raw_samples=self.n_raw_samples,
+                options={"maxiter_continuous": self.maxiter},
+                inequality_constraints=inequality_constraints,
+                equality_constraints=equality_constraints,
+                fixed_features=self.get_fixed_features(domain=domain),
+                discrete_dims={
+                    features2idx[feat.key][0]: feat.values  # type: ignore
+                    for feat in domain.inputs.get(DiscreteInput)
+                },
+                cat_dims={
+                    features2idx[feat.key][0]: feat.to_ordinal_encoding(  # type: ignore
+                        pd.Series(feat.get_allowed_categories())  # type: ignore
+                    ).tolist()
+                    for feat in domain.inputs.get(CategoricalInput)
+                    if feat.key not in fixed_keys
+                },
+            )
+        else:
+            raise ValueError(f"Unknown optimizer: {optimizer}")
 
     def get_categorical_combinations(
         self,
         domain: Domain,
-        input_preprocessing_specs: InputTransformSpecs,
     ) -> List[Dict[int, float]]:
         """Provides all possible combinations of fixed values
 
         Returns:
             list_of_fixed_features List[dict]: Each dict contains a combination of fixed values
```

Comment:
no need to duplicate type info from functin signature, prefer `list` over `List` and `dict` over `Dict`.

Reply:
Done